Assets/Scripts/UI/Canvas/ColorPicker/ColorPicker.cs

```diff
@@ -0,0 +1,70 @@
+using UnityEngine;
+
+using Framework.Enums;
+
+namespace UI.Canvas.ColorPicker
+{
+    public class ColorPicker : MonoBehaviour
+    {
+        [SerializeField] private LineRenderer routeLineRenderer;
+    
+        private void Start()
+        {
+            gameObject.SetActive(false);
+        }
+
+        /// <summary>
+        /// Checks which RouteColors enum goes with which color
+        /// </summary>
+        /// <param name="lineColor"> The RouteColors enum that gets grabbed from the button that's clicked </param>
+        /// <returns> Returns the color that is used for the ChangeColor function </returns>
+        public Color SetColor(RouteColors lineColor)
+        {
+            Color newColor = new Color();
+            
+            switch (lineColor)
+            {
+                case RouteColors.RED:
+                    newColor = new Color(1f,0f,0f);
+                    break;
+                case RouteColors.GREEN:
+                    newColor = new Color(0f, 1f, 0f);
+                    break;
+                case RouteColors.BLUE:
+                    newColor = new Color(0f, 0f, 1f);
+                    break;
+                case RouteColors.YELLOW:
+                    newColor = new Color(1f, 1f, 0f);
+                    break;
+                case RouteColors.MAGENTA:
+                    newColor = new Color(1f, 0f, 1f);
+                    break;
+                case RouteColors.ORANGE:
+                    newColor = new Color(1f, 0.5f, 0f);
+                    break;
+            }
+            
+            return newColor;
+        }
```

Comment:
```suggestion
        private Color GetColor(RouteColors lineColor)
        {
            switch (lineColor)
            {
                case RouteColors.RED:
                    return new Color(1f, 0f, 0f);
                case RouteColors.GREEN:
                    return new Color(0f, 1f, 0f);
                case RouteColors.BLUE:
                    return new Color(0f, 0f, 1f);
                case RouteColors.YELLOW:
                    return new Color(1f, 1f, 0f);
                case RouteColors.MAGENTA:
                    return new Color(1f, 0f, 1f);
                case RouteColors.ORANGE:
                    return new Color(1f, 0.5f, 0f);
                default:
                    throw new ArgumentOutOfRangeException(nameof(lineColor), lineColor, null);
            }
        }
```

1. Ik zie deze functie alleen gebruikt worden in de `ChangeColor`, dus waarom public? Vergeet niet om deze onder de andere publics te zetten dan.
2. Je maakt 2 keer een `Color` varible aan, in het begin en in de cases.
3. Set -> Get. Je zet de value niet, je pakt de value.

Reply:
Je zou mogelijk alle colors in een private list kunnen zetten en dan hier per index uit kunnen pakken. Hierdoor hoef je niet elke keer met het aan roepen hiervan een `new color` te doen.

```cs
return colorList[(int)lineColor];
```